distributed/tests/test_worker.py

```diff
@@ -2760,8 +2760,14 @@ def _acquire_replicas(scheduler, worker, *futures):
 
 def _remove_replicas(scheduler, worker, *futures):
     keys = [f.key for f in futures]
-
-    scheduler.stream_comms[worker.address].send(
+    ws = scheduler.workers[worker.address]
+    for k in keys:
+        ts = scheduler.tasks[k]
+        if ws in ts.who_has:
+            ts.who_has.remove(ws)
+            ws._nbytes -= ts.get_nbytes()
+            del ws._has_what[ts]
```

Comment:
I was wondering if this wouldn't make up a nice method on scheduler side. All three things need to be done, otherwise a validation error is raised. that would also make this test a little less hacky. Thoughts @crusaderky ?

Reply:
You mean, something like ``scheduler.remove_replica(ts, ws)``? ;) 
https://github.com/dask/distributed/pull/5341/files